src/compiler/sourcemap.ts

```diff
@@ -393,24 +391,28 @@ namespace ts {
 
                     const sourcesDirectoryPath = compilerOptions.sourceRoot ? host.getCommonSourceDirectory() : sourceMapDir;
                     const resolvedPathCache = createMap<string>();
-                    sourcemaps.calculateDecodedMappings(originalMap, (raw): void => {
+                    const absolutePathCache = createMap<string>();
+                    const sourcemapIterator = sourcemaps.decodeMappings(originalMap);
+                    for (let { value: raw, done } = sourcemapIterator.next(); !done; { value: raw, done } = sourcemapIterator.next()) {
                         // Apply offsets to each position and fixup source entries
-                        const rawPath = originalMap.sources[raw.sourceIndex];
-                        const relativePath = originalMap.sourceRoot ? combinePaths(originalMap.sourceRoot, rawPath) : rawPath;
-                        const combinedPath = combinePaths(getDirectoryPath(node.sourceMapPath!), relativePath);
-                        if (!resolvedPathCache.has(combinedPath)) {
-                            resolvedPathCache.set(combinedPath, getRelativePathToDirectoryOrUrl(
+                        if (!resolvedPathCache.has("" + raw.sourceIndex)) {

```

Comment:
Is caching the result of `"" + raw.sourceIndex` a worthwhile microoptimization?

Reply:
The VM is likely already doing that itself; but it would make the code a bit cleaner here, so I may do it for style.